src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -295,6 +295,84 @@ internal static partial class Number
                                         "80818283848586878889"u8 +
                                         "90919293949596979899"u8;
 
+        public static unsafe string FormatDecimal32(Decimal32 value, ReadOnlySpan<char> format, NumberFormatInfo info)
+        {
+            char fmt = ParseFormatSpecifier(format, out int digits);
+
+            byte* pDigits = stackalloc byte[Decimal32NumberBufferLength];
+            NumberBuffer number = new NumberBuffer(NumberBufferKind.Decimal, pDigits, Decimal32NumberBufferLength);
+
+            Decimal32ToNumber(value, ref number);
+
+            char* stackPtr = stackalloc char[CharStackBufferSize];
+            var vlb = new ValueListBuilder<char>(new Span<char>(stackPtr, CharStackBufferSize));
+
+            if (fmt != 0)
+            {
+                NumberToString(ref vlb, ref number, fmt, digits, info);
+            }
+            else
+            {
+                NumberToStringFormat(ref vlb, ref number, format, info);
+            }
+
+            string result = vlb.AsSpan().ToString();
+            vlb.Dispose();
+            return result;
+        }
+
+        public static unsafe string FormatDecimal64(Decimal64 value, ReadOnlySpan<char> format, NumberFormatInfo info)
```

Comment:
Does this need to be 3 distinct APIs or can it be a generic helper like we have for `FormatFloat<TNumber>(...) where TNumber : unmanaged, IBinaryFloatParseAndFormatInfo<TNumber>`?

At a glance it looks like the logic can be simplified and reused fairly trivially.